securedrop/tests/test_journalist.py

```diff
@@ -896,6 +896,17 @@ def test_admin_resets_hotp_with_missing_otp_secret_key(journalist_app,
     assert 'Change Secret' in resp.data.decode('utf-8')
 
 
+def test_admin_new_user_2fa_redirect(journalist_app, test_admin):
+    with journalist_app.test_client() as app:
+        _login_user(app, test_admin['username'], test_admin['password'],
+                    test_admin['otp_secret'])
+        with InstrumentedApp(journalist_app) as ins:
+            resp = app.post(
+                '/admin/2fa?uid={}'.format(test_admin['id']),
```

Comment:
What about: `url_for('admin.new_user_two_factor', uid=test_admin['id'])`? (I just think `url_for` is cleaner and the right tool for the job here). The same comment applies to the other places in the diff where this kind of string manipulation for generating query parameters is done

Reply:
When we moved to `pytest`, we removed the other calls to `url_for` because we weren't pre-pushing an app context before executing the test functions. This is something that `Flask-Testing` was doing, but it wasn't clear what would happen when there were multiple app/request contexts pushed. I can re-add this feature to the fixtures, and I think it will be fine.